src/TagsInput/TagsInput.js

```diff
@@ -0,0 +1,151 @@
+
+import { MIN_WIDTH } from './constants';
+
+type Props = {|
+  +label: string,
+  +onChangeText: (value: string) => void,
+  +placeholder: string,
+  +selected?: string[],
+  +disabled?: boolean,
+  +fontSize: number,
+  +onClearPress?: () => void,
+  +value?: string,
+|};
+
+type State = {
+  value: string,
+  containerWidth: ?number,
+  inputWidth: ?number,
+};
+
+class TagInput extends React.Component<Props, State> {
+  inputRef: any; // @TODO find a way how to type this properly, the issue with mixing react and react-native
+  static defaultProps = {
+    fontSize: 16,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      value: props.value ?? '',
+      containerWidth: null,
+      inputWidth: null,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { value } = this.props;
+    if (value !== prevProps.value) {
+      this.setState({ value });
+    }
+  };
+
+  setContainerWidth = (event: ViewLayoutEvent) => {
+    this.setState({ containerWidth: event.nativeEvent.layout.width });
+  };
+
+  setInputWidth = (event: ViewLayoutEvent) => {
+    this.setState({ inputWidth: event.nativeEvent.layout.width });
+  };
+
+  setRef = (input: React.Element<any>) => {
+    this.inputRef = input;
+  };
+
+  getWidth = (width: ?number) => {
+    const { containerWidth } = this.state;
+    if (containerWidth && width) {
+      return containerWidth - width;
+    }
+    return undefined;
+  };
+
+  handleChange = (value: string) => {
+    const { value: oldValue } = this.state;
+    const { onChangeText } = this.props;
+    if (value !== oldValue) {
+      this.setState({ value });
+      onChangeText(value);
+    }
+  };
+
+  handleClear = () => {
+    const { onClearPress } = this.props;
+    this.setState({ value: '' });
+    Platform.OS === 'web' && this.inputRef.focus();
+
+    if (onClearPress) {
+      onClearPress();
+    }
+  };
+
+  render() {
+    const { selected, fontSize, label, placeholder, disabled } = this.props;
+    const { value, inputWidth } = this.state;
+
+    return (
+      <View style={styles.container}>
+        <Text weight="bold" style={(styles.label, { fontSize })}>
+          {label}
+        </Text>
+        <View style={styles.fieldContainer} onLayout={this.setContainerWidth}>
+          <TagsContainer
+            maxWidth={this.getWidth(inputWidth)}
+            tags={selected}
+            fontSize={fontSize}
+          />
+          <InputField
+            setRef={this.setRef}
+            fontSize={fontSize}
+            disabled={disabled}
+            minWidth={MIN_WIDTH}
+            maxWidth={this.getWidth(MIN_WIDTH)}
+            onLayout={this.setInputWidth}
+            value={value}
+            placeholder={
+              selected?.length === 0 && placeholder ? placeholder : null
+            }
+            onChangeText={this.handleChange}
+          />
+        </View>
+        <DeleteButton
+          style={{ opacity: value.length > 0 ? 1 : 0 }}
```

Comment:
Can this logic be extracted above the return statement? Also, can the style be added to `styles` below? This is important for performance.